### ISSUE TITLE
Visual Studio 2022 Preview (Image classification does not work with local GPU)

### ISSUE THREAD

**Issue by enjoy8bit** [dotnet/machinelearning-modelbuilder#1751]
**Model Builder (Preview) Version**: 16.6.2.2132901
**Model Builder GPU Support Version (Preview)**: 16.6.3.2136603
**Visual Studion Version**: 2022 Version 17.0.0 Preview 3.1

**Bug description**
I cant run any local GPU image classification training, because the compatibility check always fails.

![image](https://user-images.githubusercontent.com/4288419/131555596-4208045d-5aaf-417b-872f-92347b680738.png)

It says, GPU extension is missing. - When i want to install it, it says it is already installed.
Can you help me on this, or is this simply not working on VS 2022?

**Comment by enjoy8bit** [comment]
Ok, but it says all my stuff is up 2 date:
![image](https://user-images.githubusercontent.com/4288419/131565934-9d3360b5-02d3-4653-94f9-7664b421354a.png)

![image](https://user-images.githubusercontent.com/4288419/131565989-ba85f59e-d8b8-46ce-bff7-22238deff20b.png)

Any ideas?

**Comment by beccamc** [comment]
You can download the newest build from the marketplace https://marketplace.visualstudio.com/items?itemName=MLNET.07

**Comment by LittleLittleCloud** [comment]
@beccamc 

The binary in that link is for VS2019, I don't think we release model builder for VS2022 to the public yet.

**Comment by enjoy8bit** [comment]
![image](https://user-images.githubusercontent.com/4288419/131568573-ac33c537-e36c-4b1c-8a26-7ef66eb3f149.png)

Yup, i installed that specific VS19 version. - But in VS22 i still have the issue.

**Comment by beccamc** [comment]
I don't know if that channel is still there. @enjoy8bit We are shipping an official build for VS 2022 in the next two weeks. 

@LittleLittleCloud Will we update the Marketplace build before the release on the 15th?

**Comment by LittleLittleCloud** [comment]
@beccamc 
I don't think so, but maybe we could share v16.7.3 (the one inserted in VS2022 preview4) to unblock @enjoy8bit?

**Comment by enjoy8bit** [comment]
Hello, any updates?

**Comment by beccamc** [comment]
You can download Preview 4 here. An updated Model Builder version should be built in. https://devblogs.microsoft.com/visualstudio/visual-studio-2022-preview-4-is-now-available/

**Comment by enjoy8bit** [comment]
I updated to preview 4.
I still get this message:
![image](https://user-images.githubusercontent.com/4288419/133493833-8f23df39-f745-4a4f-aa1d-9b42667203d1.png)

So where do i get correct version of ML.NET Model Builder GPU Support for VS2022 ? - Many thanks in advance

**Comment by LittleLittleCloud** [comment]
After updating your VS 2022 to preview 4, The version of your ModelBuilder extension should be `16.7.3.2143002`. So you can just update your GPU extension (which was `16.6.3.2136603`) to the recent updated one (`16.7.3.2143002`) in the [market place](https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilderGPU) and everything should work.

**Comment by LittleLittleCloud** [comment]
> After updating your VS 2022 to preview 4, The version of your ModelBuilder extension should be `16.7.3.2143002`. So you can just update your GPU extension (which was `16.6.3.2136603` according to previous conversation) to the recent updated one (`16.7.3.2143002`) in the [market place](https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilderGPU) and everything should work.

Just found out that the GPU extension on the market place is for VS2019 only, sorry for the inconvenience.

@beccamc 
Didl we release GPU extension for VS2022. If it's not released yet, maybe we can share a download link to unpublished build to unblock @enjoy8bit ?

**Comment by enjoy8bit** [comment]
I am starting see a pattern or a Déjà vu in here :)

**Comment by pierre01** [comment]
Sucky support for ML.NET, Why can't it be at least backward compatible on a technology that isn't even  2 years old.

**Comment by enjoy8bit** [comment]
> @pierre01 @enjoy8bit
> 
> Looks like a bug in re-entrance... Could you try creating a new model builder for image classifcation and see if error goes? Or instead, you could open your .mbconfig file as json and add a `FolderPath` property with full path to your image folder in `FolderSource`
> 
> ```json
> "TrainingTime": 2147482,
>   "Scenario": "ImageClassification",
>   "DataSource": {
>     "Type": "Folder",
>     "Version": 1,
>     "FolderPath": "add/your/full/path/to/folder/here"
>   },
> ```
> 
> @zewditu Could you take a look at this? The exception is thrown when sending telemetry,, which shouldn't. Telemetry should never send exception and FolderPath should never missing in DataSource...

![image](https://user-images.githubusercontent.com/4288419/138336981-051f8ae8-8ba5-4194-8d27-f67be781f134.png)

The full path seems to already be used here.

**Comment by pierre01** [comment]
No:
***@***.***
 And when I trey to reinstall the

ML.NET Model Builder GPU Support 2022

I get:

***@***.***

**Comment by beccamc** [comment]
Sorry @pierre01 I can't see the image or error you included after "I get:"

**Comment by pierre01** [comment]
> Sorry @pierre01 I can't see the image or error you included after "I get:"

I modified the images in my comment above. 

**You can ignore my comment as the latest version has fixed this issue .**

**Comment by beccamc** [comment]
HI all, closing this as I believe the issue is resolved. If anyone encounters this, make sure that GPU and Model Builder extension versions match. 

If there are any other GPU issues please re-open or file a new issue. Thanks!
